pyramid_openapi3/tests/test_validation.py

```diff
@@ -0,0 +1,255 @@
+        """unittest.TestCase tearDown for each test method.
+
+        Tear down everything set up in setUp.
+        """
+        tearDown()
+        self.config = None
+
+    def _add_view(self, view_func=None, openapi=True) -> None:
+        """Add a simple example view.
+
+        :param view_func: an optional view callable.
+        :param openapi: if True, enable openapi view deriver
+        """
+        self.config.add_route("foo", "/foo")
+        if not view_func:
+            view_func = lambda *arg: "foo"  # noqa: E731  # pragma: no branch
+        self.config.add_view(
+            openapi=openapi, renderer="json", view=view_func, route_name="foo"
+        )
+
+    def _add_default_exception_view(self) -> None:
+        """Install Pyramid default_exception_response_view."""
+        # setup default exception response view
+        from pyramid.httpexceptions import default_exceptionresponse_view
+        from pyramid.interfaces import IExceptionResponse
+
+        self.config.add_view(
+            view=default_exceptionresponse_view, context=IExceptionResponse
+        )
+
+    def _get_view(self) -> View:
+        """Return wrapped view method registered in _add_view."""
+        request_interface = self.config.registry.queryUtility(IRouteRequest, name="foo")
+        view = self.config.registry.adapters.registered(
+            (IViewClassifier, request_interface, Interface), IView, name=""
+        )
+        return view
+
+    def _get_request(self, params: t.Dict = None) -> DummyRequest:
+        """Create a DummyRequest instance matching example view.
+
+        :param params: Query parameter dictionary
+        """
+        request = DummyRequest(config=self.config, params=params)
+        request.matched_route = DummyRoute(name="foo", pattern="/foo")
+        return request
+
+    def test_request_validation_error(self) -> None:
+        """Test View raises RequestValidationError.
+
+        Request parameters don't match schema.
+        """
+        self._add_view()
+        view = self._get_view()
+        request = self._get_request()
+        from pyramid_openapi3 import RequestValidationError
+
+        with self.assertRaises(RequestValidationError) as cm:
+            view(None, request)
+        response = cm.exception
+        self.assertEqual(str(cm.exception), "Missing required parameter: bar")
+        # not enough of pyramid has been set up so we need to render the
+        # exception response ourselves.
+        response.prepare({"HTTP_ACCEPT": "application/json"})
+        self.assertIn("Missing required parameter: bar", response.json["message"])
+
+    def test_view_raises_valid_http_exception(self) -> None:
+        """Test View raises HTTPException.
+
+        Example view raises a defined response code.
+        """
+        from pyramid.httpexceptions import HTTPBadRequest
+
+        self._add_view(
+            lambda *arg: (_ for _ in ()).throw(HTTPBadRequest("bad foo request"))
```

Comment:
Is the generator `throw` here meaningful, or is it a workaround for not being able to `raise` inside a `lambda`? I would think that a normal function would be a clearer analogue to a normal view.

Reply:
Agree, took me a while myself to understand this `generator throw` hack. 
Wasn't sure what to do. Either use this construct to raise an exception in a `lambda` or break the pattern used in other tests and use a local function. (updated pull request)